ConcurrencyApp/UI/FetchPostsViewController.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  ViewController.swift
+//  ConcurrencyApp
+//
+//  Created by Renato F. dos Santos Jr on 27/06/25.
+//
+
+import UIKit
+import SwiftUI
+
+final class FetchPostsViewController: UIViewController {
+    private let contentView: FetchPostsView
+    private let service: FetchPostsService
+
+    init(contentView: FetchPostsView = FetchPostsView(),
+        service: FetchPostsService = FetchPostsService()) {
+        self.contentView = contentView
+        self.service = service
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    deinit {
+        print("\(self) deinitialized")
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Lifecycle
+    override func loadView() {
+        view = contentView
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        title = "ConcurrencyApp"
+        contentView.delegate = self
+    }
+}
+
+extension FetchPostsViewController: FetchPostsViewButtonDelegate {
+    func didTapButton(_ contentView: FetchPostsView) {
+        print("didTapButton on contentView: \(contentView)")
+        contentView.update(with: .loading("Fetching data..."))
+
+        let queue = DispatchQueue(label: "com.renato.example.fetchQueue")
```

Comment:
How about we move this queue to the variable declaration? Like that, you gave complete control of the thread to the ViewController. But loved to see you using queues 🤩